src/highrise/webapi.py

```diff
@@ -0,0 +1,135 @@
+from os import environ
+from typing import Any, Literal, Type
+
+from aiohttp import ClientSession
+from cattrs import Converter
+from pendulum import DateTime, parse
+
+from .models_webapi import (
+    GetPublicPostResponse,
+    GetPublicPostsResponse,
+    GetPublicRoomResponse,
+    GetPublicRoomsResponse,
+    GetPublicUserResponse,
+    GetPublicUsersResponse,
+)
+
+converter = Converter()
+converter.register_structure_hook(DateTime, lambda ts, _: parse(ts))
+
+SORT_OPTION = Literal["desc", "asc"]
+
+
+class WebAPI:
+    """A class for interacting with the Highrise Web API.
+
+    This class provides asynchronous methods for fetching data from the Highrise API.
+    It supports fetching single and multiple users, rooms, and posts.
+
+    Each method corresponds to a specific endpoint on the Highrise API, and returns a
+    structured response based on the response JSON and webapi models.
+    """
+
+    url: str = environ.get("HR_WEBAPI_URL", "https://rc-webapi.highrise.game")
+
+    async def get_user(self, user_id: str) -> GetPublicUserResponse:
+        """Fetch a single user given its user_id.
+
+        Args:
+            user_id (str): The unique identifier for a user.
+
+        Returns:
+            GetPublicUserResponse: The public data of the user.
+        """
+        endpoint = f"/users/{user_id}"
+        return await self.send_request(endpoint, GetPublicUserResponse)
+
+    async def get_users(
+        self,
+        starts_after: str = "",
+        ends_before: str = "",
+        sort_order: SORT_OPTION = "desc",
+        limit: int = 20,
+        username: str = "",
+    ) -> GetPublicUsersResponse:
+        """Fetch a list of users, can be filtered, ordered, and paginated.
+
+        Returns:
+            GetPublicUsersResponse: A list of public data of users.
+        """
+        endpoint = f"/users?starts_after={starts_after}&ends_before={ends_before}&sort_order={sort_order}&limit={limit}&username={username}"
```

Comment:
Can we make this a bit nicer, so that we don't send empty strings if for example username is not set for all the endpoint making.